src/utils/item.ts

```diff
@@ -130,3 +136,41 @@ export const paginationContentFilter = (
   items
     .filter((i) => i.type !== ItemType.FOLDER)
     .filter((i) => !i.settings?.isPinned);
+
+// handle item children tree
+let matchingIds: string[] = [];
+const handleItemChildren = (
+  data: DiscriminatedItem[],
+  obj: DiscriminatedItem & { children?: DiscriminatedItem[] },
+) => {
+  if ((obj?.extra as FolderItemExtra)?.folder?.childrenOrder?.length) {
```

Comment:
I think it's better to have a first check `if(item.type === ItemType.FOLDER)`. Maybe then the types will resolve themselves.